x-pack/plugins/index_management/public/application/components/mappings_editor/mappings_editor.tsx

```diff
@@ -120,6 +128,40 @@ export const MappingsEditor = React.memo(
       [dispatch]
     );
 
+    const [isLicenseCheckComplete, setIsLicenseCheckComplete] = useState(false);
+    useEffect(() => {
+      const subscription = licensing?.license$.subscribe((license: ILicense) => {
```

Comment:
what do you think about extracting the license check subcription check into the plugin level and just pass something like `canUseSyntheticOption` variable into the `renderApp` method? that way we dont have tightly coupled logic and if the license condition changes its easier to update (I think we did something similar for checking if the user can use the new api key trust model in remote clusters)

Reply:
That's a great idea! This might actually help with testing if the synthetic option is displayed or not based on the license. I think the issues that I face now might be due to the fact that in the initial render the check is not complete and so the synthetic option is not displayed even when it should.